### PR TITLE
Fix #4557: Handle New with HKTypeLambda alias

### LINKED ISSUE #4557: Inheritance through a parameterized type alias fails
```scala
class Parser[T, Elem]

object Test {
  type P[T] = Parser[T, Char]
  class CustomParser extends P[Unit]
}
```

Dotty wrongly complains that `Test.P is not a class type`

### PR DESCRIPTION
**Pull request by Medowhill**

### CODE REVIEW

**Comment by allanrenucci** [comment]
You can sign the CLA at https://www.lightbend.com/contribute/cla/scala

**Comment by Medowhill** [comment]
Thank you for the link! I signed the CLA.

**Comment by allanrenucci** [comment]
[Confirmed](https://www.lightbend.com/contribute/cla/scala/check/Medowhill). Don't worry about the bot, it's pretty buggy

File: tests/pos/i4557b.scala [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  type T2[T, S] = C2[T, S]
+  class DC2 extends C2[Int, String](0, "a")
+  class DT2 extends T2[Int, String](1, "b")
+
+  type T21[T] = C2[T, String]
+  class DT21 extends T21[Int](2, "c")
+
+  type T22 = C2[Int, String]
+  class DT22 extends T22(3, "d")
```
Comment (allanrenucci) [inline tests/pos/i4557b.scala]: Can you also add tests with multiple alias. E.g:
```scala
type T2[T, S] = C2[T, S]
type TT2[T, S] = T2[T, S]
class DTT22 extends TT2(3, "d")
```

File: tests/pos/i4557b.scala [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+      new DC0,
+      new DT0,
+      new DC1,
+      new DT1,
+      new DT11,
+      new DC2,
+      new DT2,
+      new DT21,
+      new DT22
+    ).foreach(_.f)
```
Comment (allanrenucci) [inline tests/pos/i4557b.scala]: I you intended to make this a run test, it should go under `test/run` and have a `main` method in a `class Test`.
You can either use assertions or a check file to test the output of a run test

**Comment by allanrenucci** [comment]
Here is a link to the lost [commit](https://github.com/Medowhill/dotty/commit/9b9e6d551ec826e5d31fce9ec732355817029d04)

**Pull request closed without merging**